src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -294,28 +292,28 @@ void c_typecheck_baset::typecheck_expr_main(exprt &expr)
   {
     // op0 is a declaration,
     // op1 the bound expression
-    assert(expr.operands().size()==2);
     expr.type()=bool_typet();
+    auto &binary_expr = to_binary_expr(expr);
 
-    if(expr.op0().get(ID_statement)!=ID_decl)
+    if(binary_expr.op0().get(ID_statement) != ID_decl)
     {
       error().source_location = expr.source_location();
       error() << "expected declaration as operand of quantifier" << eom;
       throw 0;
     }
 
-    if(has_subexpr(expr.op1(), ID_side_effect))
+    if(has_subexpr(binary_expr.op1(), ID_side_effect))
     {
       error().source_location = expr.source_location();
       error() << "quantifier must not contain side effects" << eom;
       throw 0;
     }
 
     // replace declaration by symbol expression
-    symbol_exprt bound=to_symbol_expr(expr.op0().op0());
-    expr.op0().swap(bound);
+    symbol_exprt bound = to_symbol_expr(to_unary_expr(binary_expr.op0()).op());
```

Comment:
Shouldnt't this be `to_code_decl(binary_expr.op0()).symbol()`?

Reply:
`to_code_decl(to_code(binary_expr.op0())).symbol()`
Fixed